src/PyBytesProxyHandler.cc

```diff
@@ -0,0 +1,414 @@
+/**
+ * @file PyBytesProxyHandler.cc
+ * @author Philippe Laporte (philippe@distributive.network)
+ * @brief Struct for creating JS Uint8Array-like proxy objects for immutable bytes objects
+ * @date 2024-07-23
+ *
+ * @copyright Copyright (c) 2024 Distributive Corp.
+ *
+ */
+
+
+#include "include/PyBytesProxyHandler.hh"
+
+#include <jsapi.h>
+#include <js/ArrayBuffer.h>
+
+
+const char PyBytesProxyHandler::family = 0;
+
+
+static bool array_valueOf(JSContext *cx, unsigned argc, JS::Value *vp) {
+  JS::CallArgs args = JS::CallArgsFromVp(argc, vp);
+
+  JS::RootedObject proxy(cx, JS::ToObject(cx, args.thisv()));
+  if (!proxy) {
+    return false;
+  }
+
+  JS::PersistentRootedObject* arrayBuffer = JS::GetMaybePtrFromReservedSlot<JS::PersistentRootedObject>(proxy, OtherSlot);
+  JS::RootedObject rootedArrayBuffer(cx, arrayBuffer->get());
+
+  auto byteLength = JS::GetArrayBufferByteLength(rootedArrayBuffer);
+
+  bool isSharedMemory; 
+  JS::AutoCheckCannotGC autoNoGC(cx);
+  uint8_t *data = JS::GetArrayBufferData(rootedArrayBuffer, &isSharedMemory, autoNoGC);
+
+  const size_t STRING_LENGTH = byteLength*2 - 1;
+  JS::Latin1Char* buffer = (JS::Latin1Char *)malloc(sizeof(JS::Latin1Char) * STRING_LENGTH);
+
+  buffer[0] = data[0];
+  for (Py_ssize_t index = 1; index < byteLength; index++) {
+    buffer[index*2 - 1] = ',';
+    buffer[index*2] = data[index];
+  }
+
+  JS::UniqueLatin1Chars str(buffer);
+  args.rval().setString(JS_NewLatin1String(cx, std::move(str), STRING_LENGTH));
+  return true;
```

Comment:
```suggestion
  size_t numberOfDigits = 0;
  for (size_t i = 0; i < byteLength; i++) {
    numberOfDigits += data[i] < 10 ? 1 : data[i] < 100 ? 2 : 3; 
  }
  const size_t STRING_LENGTH = byteLength + numberOfDigits;
  JS::Latin1Char* buffer = (JS::Latin1Char *)malloc(sizeof(JS::Latin1Char) * STRING_LENGTH);
  
  size_t charIndex = 0;
  sprintf((char*)&buffer[charIndex], "%d", data[0]);
  charIndex += data[0] < 10 ? 1 : data[0] < 100 ? 2 : 3;
  for (size_t dataIndex = 1; dataIndex < byteLength; dataIndex++) {
    buffer[charIndex] = ',';
    charIndex++;
    sprintf((char*)&buffer[charIndex], "%d", data[dataIndex]);
    charIndex += data[dataIndex] < 10 ? 1 : data[dataIndex] < 100 ? 2 : 3;
  }

  JS::UniqueLatin1Chars str(buffer);
  args.rval().setString(JS_NewLatin1String(cx, std::move(str), STRING_LENGTH - 1)); // don't include null byte
  return true;
```

This should make a string of numbers rather than a string of characters, sorry about that. Could you write some tests for these (the ones in array_methods)